zerver/lib/emoji.py

```diff
@@ -35,6 +36,14 @@
     + f"))(?![^{terminal_symbols}])"
 )
 
+
+def data_url() -> str:
+    # This bakes a hash into the URL, which looks something like
+    # static/webpack-bundles/files/64.0cdafdf0b6596657a9be.png
+    # This is how Django deals with serving static files in a cacheable way. See PR #22275 for details.
+    return staticfiles_storage.url("generated/emoji/emoji_api.json")
```

Comment:
Would it be feasible to check that the JSON file exists, and if not, give `null` in the `/register` response?

From local testing, it seems possible to have a running server that doesn't have the JSON file (I guess I forgot to run `tools/setup/emoji/build_emoji`, which creates the file). And with that being the case, I received `http://10.10.10.29:9991/static/generated/emoji/emoji_api.json` for the `emoji_data_url`. Requesting that in my browser on my computer, I got this:

<img width="605" alt="image" src="https://user-images.githubusercontent.com/22248748/181373131-8b9e73e5-9ba2-4f15-8811-0183d19a892d.png">

Reply:
Yay, I ran `tools/setup/emoji/build_emoji`, and now, as expected, I get a bunch of JSON instead of that "Page not found" page.

I'm still getting `emoji_data_url: "http://10.10.10.29:9991/static/generated/emoji/emoji_api.json"` in the register response. Should I expect a URL with a hash in it?

Reply:
I think the development environment doesn't do URL hashing, so that seems expected.

Reply:
Ah OK, got it. I'm still wondering about this:

> Would it be feasible to check that the JSON file exists, and if not, give `emoji_data_url: null` in the `/register` response? Or else require running servers to have the JSON file, maybe.

Reply:
I can't imagine a situation where the JSON file wouldn't exist, since if `build_emoji` wasn't run then we would also have no emoji, and the comment where the file is written to mentions it's a public API endpoint so I wouldn't expect it to be removed. 

I could add a check somewhere, though, if anyone thinks we need it! I'm not totally sure where the best place would be. Maybe @timabbott would know?

Reply:
Ah okay so I think what happened is we didn't force a re-provision when the `build_emoji` script changed, so you had run the old one but hadn't gotten the updates. Would this be relevant to servers that we'd be pushing updates to as well? Maybe we need to flip some switch that shows that this commit requires a reprovision (or at least to rebuild emoji).

Reply:
Ah got it, thanks! This sounds like a productive direction that'll help make sure servers serve the data where they say they will.

Tim, if you're skimming this thread, you can go straight to Evy's questions about forcing a re-provision, just above.

Reply:
You should increase `PROVISION_VERSION` to notify users of the development environment that they need to re-run `provision`, which will then automatically detect that `build_emoji` changed and run it. A minor `PROVISION_VERSION` bump suffices because you’re only adding stuff, so developers going backwards through history don’t need to run it again to undo the change.

This is not an issue for production, where `update-prod-static` always runs `build_emoji`.

Reply:
perfect, thanks!